torch/csrc/autograd/function.h

```diff
@@ -419,6 +429,7 @@ struct TORCH_API Node : std::enable_shared_from_this<Node> {
 
   // Topological ordering of nodes
   uint64_t topological_nr_ = 0;
+  mutable bool node_has_been_used = 0;
```

Comment:
Temporary change for testing only - we should eventually put this behind a `#ifndef NDEBUG`